arroyo-api/src/connection_tables.rs

```diff
@@ -86,45 +98,235 @@ async fn get_and_validate_connector<E: GenericClient>(
+        .bind(&client, &auth_data.organization_id, &pub_id)
+        .await
+        .map_err(|e| handle_delete("connection_table", "pipelines", e))?;
+
+    if deleted == 0 {
+        return Err(ErrorResp {
+            status_code: StatusCode::NOT_FOUND,
+            message: "Connection table not found".to_string(),
+        });
+    }
+
+    Ok(())
 }
 
-pub(crate) async fn create(
-    req: CreateConnectionTableReq,
-    auth: AuthData,
-    pool: &Pool,
-) -> Result<(), Status> {
-    let mut c = pool.get().await.map_err(log_and_map)?;
+/// Test a Connection Table
+#[utoipa::path(
+    post,
+    path = "/v1/connection_tables/test",
+    tag = "connection_tables",
+    request_body = ConnectionTablePost,
+    responses(
+        (status = 200, description = "Job output as 'text/event-stream'"),
+    ),
+)]
+pub(crate) async fn test_connection_table(
+    State(state): State<AppState>,
+    bearer_auth: BearerAuth,
+    WithRejection(Json(req), _): WithRejection<Json<ConnectionTablePost>, ApiError>,
+) -> Result<Sse<impl Stream<Item = Result<Event, Infallible>>>, ErrorResp> {
+    let client = client(&state.pool).await?;
+    let auth_data = authenticate(&state.pool, bearer_auth).await?;
+
+    let (connector, _, config, schema) =
+        get_and_validate_connector(&req, &auth_data, &client).await?;
+
+    let (tx, mut rx) = channel(8);
+
+    let (tx2, rx2) = channel(8);
+
+    connector
+        .test(&req.name, &config, &req.config, schema.as_ref(), tx)
+        .map_err(|e| {
+            Status::invalid_argument(format!("Failed to parse config or schema: {:?}", e))
+        })?;
+
+    tokio::spawn(async move {
+        loop {
+            let m = match rx.recv().await {
+                Some(d) => d,
+                None => {
+                    warn!("Stream closed");
+                    break;
+                }
+            };
+
+            let message = match m {
+                Ok(msg) => msg,
+                Err(_) => {
+                    error!("Stream error");
+                    break;
+                }
+            };
+
+            let e: Result<Event, Infallible> =
+                Ok(Event::default().data(serde_json::to_string(&message).unwrap()));
+            if tx2.send(e).await.is_err() {
+                break;
+            }
+
+            if message.done {
+                break;
+            }
+        }
+    });
+
+    Ok(Sse::new(ReceiverStream::new(rx2)))
+}
+
+fn get_connection_profile(
+    c: &DbConnectionTable,
+    connector: &dyn ErasedConnector,
+) -> Option<ConnectionProfile> {
+    let config = serde_json::to_string(&c.connection_config.as_ref()?).unwrap();
+    Some(ConnectionProfile {
+        id: c.pub_id.clone(),
+        name: c.connection_name.as_ref()?.clone(),
+        connector: c.connection_type.as_ref()?.clone(),
+        description: connector.config_description(&config).ok()?,
+        config,
+    })
+}
+
+pub(crate) async fn get<C: GenericClient>(
```

Comment:
Do we still need this method? It looks like it's used by pipelines but could that rely on the new get_connection_tables instead?

Reply:
I renamed this and made `get_connection_tables` use it.